yum-plugin/test/com/tw/go/plugin/material/artifactrepository/yum/exec/HttpConnectionCheckerTest.java

```diff
@@ -1,96 +1,104 @@
-        GetMethod getMethod = mock(GetMethod.class);
-        when(checker.getGetMethod(Matchers.<String>any())).thenReturn(getMethod);
-        when(httpClient.executeMethod(getMethod)).thenReturn(HttpStatus.SC_OK);
+    public void shouldNotThrowExceptionIfCheckConnectionToTheRepoPasses() throws Exception {
+        webServer.enqueue(new MockResponse().setResponseCode(200).setBody(""));
+        HttpUrl url = webServer.url("/repodata/repomd.xml");
 
-        checker.checkConnection("url", new Credentials(null, null));
+        checker.checkConnection(url.toString(), new Credentials(null, null));
 
-        verify(httpClient).executeMethod(getMethod);
-        verify(getMethod, never()).getStatusLine();
+        RecordedRequest recordedRequest = webServer.takeRequest();
+        assertEquals("/repodata/repomd.xml", recordedRequest.getPath());
+        assertEquals(1, webServer.getRequestCount());
     }
 
     @Test
+    public void shouldPerformBasicAuthUsingChallengeResponseAuth() throws Exception {
+        webServer.enqueue(new MockResponse().setResponseCode(401).setHeader("WWW-Authenticate", "Basic realm=\"YumRepo\""));
+        webServer.enqueue(new MockResponse().setResponseCode(200).setBody(""));
+
+        HttpUrl url = webServer.url("/repodata/repomd.xml");
+
+        checker.checkConnection(url.toString(), new Credentials("foo", "bar"));
+        assertEquals(2, webServer.getRequestCount());
+
+        RecordedRequest recordedRequest = webServer.takeRequest();
+        assertEquals("/repodata/repomd.xml", recordedRequest.getPath());
+        assertThat(recordedRequest.getHeaders().get("Authorization"), is((String) null));
+
+        recordedRequest = webServer.takeRequest();
+        assertEquals("/repodata/repomd.xml", recordedRequest.getPath());
+        assertThat(recordedRequest.getHeaders().get("Authorization"), is("Basic Zm9vOmJhcg=="));
+    }
+
+
+    @Test
     public void shouldFailCheckConnectionToTheRepoWhenHttpClientReturnsAUnSuccessfulReturnCode() throws IOException {
-        HttpClient httpClient = mock(HttpClient.class);
-        GetMethod getMethod = mock(GetMethod.class);
-        when(checker.getHttpClient()).thenReturn(httpClient);
-        when(checker.getGetMethod(Matchers.<String>any())).thenReturn(getMethod);
-        when(httpClient.executeMethod(getMethod)).thenReturn(HttpStatus.SC_UNAUTHORIZED);
-        when(getMethod.getStatusLine()).thenReturn(new StatusLine("HTTP/1.1 401 Unauthorized"))    ;
+        webServer.enqueue(new MockResponse().setResponseCode(500));
+        HttpUrl url = webServer.url("/repodata/repomd.xml");
 
         try {
-            checker.checkConnection("url", new Credentials(null, null));
+            checker.checkConnection(url.toString(), new Credentials(null, null));
             fail("should fail");
         } catch (Exception e) {
-            assertThat(e.getMessage(), is("HTTP/1.1 401 Unauthorized"));
+            assertThat(e.getMessage(), is("HTTP/1.1 500 Server Error"));
```

Comment:
Yes, the existing test was slightly incorrect, 401 could indicate a challenge response, and I wanted another test to handle an unsuccessful response.

I guess there's a need for a test that verifies behavior of 401 status without a challenge — I'll add one in.